app/sitemap.ts

```diff
@@ -0,0 +1,33 @@
+import { getCollections, getPages, getProducts } from 'lib/shopify';
+import { MetadataRoute } from 'next';
+
+const baseUrl = process.env.NEXT_PUBLIC_VERCEL_URL
+  ? `https://${process.env.NEXT_PUBLIC_VERCEL_URL}`
```

Comment:
Can this be done dynamically like this or should we hardcode it to `demo.vercel.store`?

Reply:
This should work, vs. hardcoding. There isn't a `metabaseBase` here similar to the built-in SEO support AFAIK, but could be wrong.